docs/DeveloperGuide.md

```diff
@@ -1,33 +1,244 @@
+method accordingly. The `addBunny` command calls the `parseSingleCharacterTaggedParamsFromUserInput` method from the `Parsers` class to extract the `idea` and `genre` arguments 
+from the command. These are then used to create a new `Bunny` object that is then added to the `bunniesList` ArrayList. The `addBunnyMessage` method from `UI` is then called
+to print the message that the `Bunny` idea object has been sucessfully added to the ArrayList.
+
+## Names class family
+
+![Names UML Class Diagram](graphics/diagrams/classDiagram_Names.png)
+<center><i>Figure 8: Names UML Class Diagram</i></center>
+
+The above class diagram (Figure 8) describes the overall architecture of the name list functionalities. The Names class 
+has the protected ArrayList of names, nameList, that is accessed by the Names class method getName which randomly gets 
+a selected name from the nameList ArrayList. Similarly, nameList is also accessed by the Names class which contains the 
+filterNames function which can filter through the list and obtain names with specified keywords using the command filter 
+name <NAME>, where the user may choose to omit the NAME when running the command. Similarly, nameList is also accessed 
+by the Names class which contains the listNames function which displays all the names stored in the nameList ArrayList. 
+This is the same as the filterNames function when given no input String. Similarly, nameList is also accessed by the 
+Names class which contains the addName function which adds a name to the list of names stored in the nameList ArrayList 
+using the command add name <NAME>. The NAME cannot be omitted. Similarly, nameList is also accessed by the Names class 
+which contains the deleteName function which removes a name from the list of names stored in the nameList ArrayList. 
+The command to do this deletes name <INDEX>. The INDEX cannot be omitted and the range of the INDEX can be determined 
+from the listNames function above.
+
+The NamesDB class accesses the nameList and overwrites the current Names.txt file in the data directory, saving all 
+String objects in nameList into the file using the updateDB method. String objects saved in that file can then be read 
+by the NamesDB class and saved into the nameList ArrayList using the loadDB method. In the event of the database 
+Names.txt not existing, the NamesDB class will create the Names.txt database and populate the database with 500 names using the loadDB method.
+
+As shown in Figure 8, both the NamesDB class and the Names class will create the NameException class. This is a subclass 
+that inherits from the Exception superclass and passes the exception message to the superclass. In the event of an 
+exception, it is thrown from the methods in NamesDB class and Names class and handled by the NameException class.
```

Comment:
Similar to the above comment, I think you can try using a different font to differentiate between explanation and references to class/objects? This could make it more readable and also coherent with the rest of the doc. :)

Reply:
I agree, I think the methods should be in the `methodname` format which was used above too.